2025-02-17 22:11:26,976 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:11:28,515 ERROR: Exception on /dashboard [GET] [in /home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/app/routes.py", line 58, in dashboard
    paychecks_json = json.dumps(paychecks_data)
                     ^^^^
NameError: name 'json' is not defined. Did you forget to import 'json'?
2025-02-17 22:11:28,517 ERROR: Server Error: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application. [in /home/josh/BudgetApp/app/errors.py:19]
2025-02-17 22:12:30,851 INFO: Finance app startup [in /home/josh/BudgetApp/app/__init__.py:48]
2025-02-17 22:12:33,882 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:12:38,229 ERROR: Exception on /dashboard [GET] [in /home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/app/routes.py", line 58, in dashboard
    paychecks_json = json.dumps(paychecks_data)
                     ^^^^
NameError: name 'json' is not defined. Did you forget to import 'json'?
2025-02-17 22:12:38,232 ERROR: Server Error: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application. [in /home/josh/BudgetApp/app/errors.py:19]
2025-02-17 22:12:38,282 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:12:56,134 INFO: Finance app startup [in /home/josh/BudgetApp/app/__init__.py:48]
2025-02-17 22:13:02,630 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:13:16,174 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:13:17,728 ERROR: Exception on /dashboard [GET] [in /home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/app/routes.py", line 58, in dashboard
    paychecks_json = json.dumps(paychecks_data)
                     ^^^^
NameError: name 'json' is not defined. Did you forget to import 'json'?
2025-02-17 22:13:17,731 ERROR: Server Error: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application. [in /home/josh/BudgetApp/app/errors.py:19]
2025-02-17 22:15:33,352 INFO: Finance app startup [in /home/josh/BudgetApp/app/__init__.py:48]
2025-02-17 22:15:36,908 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:15:41,877 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:15:43,645 ERROR: Exception on /dashboard [GET] [in /home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/app/routes.py", line 60, in dashboard
    paychecks_json=jsonify(paychecks_data).get_data(as_text=True),
                   ^^^^^^^
NameError: name 'jsonify' is not defined
2025-02-17 22:15:43,648 ERROR: Server Error: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application. [in /home/josh/BudgetApp/app/errors.py:19]
2025-02-17 22:18:01,471 INFO: Finance app startup [in /home/josh/BudgetApp/app/__init__.py:48]
2025-02-17 22:18:04,918 ERROR: Page not found: 404 Not Found: The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again. [in /home/josh/BudgetApp/app/errors.py:12]
2025-02-17 22:18:12,087 ERROR: Exception on /dashboard [GET] [in /home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py:825]
Traceback (most recent call last):
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 1455, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 869, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 867, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask/app.py", line 852, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/venv/lib/python3.13/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/home/josh/BudgetApp/app/routes.py", line 60, in dashboard
    paychecks_json=json.dumps(paychecks_data),
                   ^^^^
NameError: name 'json' is not defined. Did you forget to import 'json'?
2025-02-17 22:18:12,089 ERROR: Server Error: 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application. [in /home/josh/BudgetApp/app/errors.py:19]
